src/libcore/iter/mod.rs

```diff
@@ -520,6 +520,35 @@ impl<I> Iterator for Cycle<I> where I: Clone + Iterator {
 #[unstable(feature = "fused", issue = "35602")]
 impl<I> FusedIterator for Cycle<I> where I: Clone + Iterator {}
 
+/// An iterator that skips n elements every iteration.
+///
+/// This `struct` is created by the [`every_nth`] method on [`Iterator`]. See
+/// its documentation for more.
+///
+/// [`every_nth`]: trait.Iterator.html#method.every_nth
+/// [`Iterator`]: trait.Iterator.html
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+#[stable(feature = "rust1", since = "1.0.0")]
+#[derive(Clone, Debug)]
+pub struct EveryNth<I> {
+    iter: I,
+    step: usize,
+}
+
+#[stable(feature = "rust1", since = "1.0.0")]
+impl<I> Iterator for EveryNth<I> where I: Iterator {
+    type Item = I::Item;
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        let elt = self.iter.next();
+        if self.step > 0 {
+            self.iter.nth(self.step - 1);
```

Comment:
@johncf I think a similar rationale applies to [`Peekable`](https://doc.rust-lang.org/std/iter/struct.Peekable.html). It would be easier to write it so that it always contains the current element (or `None`). Instead it was explicitly coded to keep track of whether the element has been pulled from the inner iterator.